R/data_rename.R

```diff
@@ -10,18 +10,40 @@
 #'   pipe-workflow.
 #'
 #' @param data A data frame, or an object that can be coerced to a data frame.
-#' @param pattern Character vector. For `data_rename()`, indicates columns that
-#'   should be selected for renaming. Can be `NULL` (in which case all columns
-#'   are selected). For `data_addprefix()` or `data_addsuffix()`, a character
-#'   string, which will be added as prefix or suffix to the column names. For
-#'   `data_rename()`, `pattern` can also be a named vector. In this case, names
-#'   are used as values for the `replacement` argument (i.e. `pattern` can be a
-#'   character vector using `<new name> = "<old name>"` and argument `replacement`
-#'   will be ignored then).
-#' @param replacement Character vector. Indicates the new name of the columns
-#'   selected in `pattern`. Can be `NULL` (in which case column are numbered
-#'   in sequential order). If not `NULL`, `pattern` and `replacement` must be
-#'   of the same length. If `pattern` is a named vector, `replacement` is ignored.
+#' @param pattern Character vector.
+#'   - For `data_addprefix()` or `data_addsuffix()`, a character string, which
+#'     will be added as prefix or suffix to the column names.
+#'   - For `data_rename()`, indicates columns that should be selected for
+#'     renaming. Can be `NULL` (in which case all columns are selected).
+#'     `pattern` can also be a named vector. In this case, names are used as
+#'     values for the `replacement` argument (i.e. `pattern` can be a character
+#'     vector using `<new name> = "<old name>"` and argument `replacement` will
+#'     be ignored then).
+#' @param replacement Character vector. Can be one of the following:
+#'   - A character vector that indicates the new names of the columns selected
+#'     in `pattern`. `pattern` and `replacement` must be of the same length.
+#'   - `NULL`, in which case columns are numbered in sequential order.
+#'   - A string (i.e. character vector of length 1) with a "glue" styled pattern.
+#'     Currently supported tokens are:
+#'     - `{col}` (or `{name}`), which will be replaced by the column name, i.e.
+#'       the corresponding value in `pattern`.
```

Comment:
I'd rather keep only one of them (but no preference between the two). This argument can already take various inputs, I don't think we should add aliases in possible values on top of that.